corehq/apps/export/models/new.py

```diff
@@ -1989,13 +1995,14 @@ def get_latest_export_schema(domain, app_id, case_type):
 
     @classmethod
     def _process_app_build(cls, current_schema, app, case_type):
-        case_property_mapping = get_case_properties(
-            app,
-            [case_type],
+        builder = ParentCasePropertyBuilder(
+            app.domain,
+            [app],
             include_parent_properties=False
         )
-        parent_types = (ParentCasePropertyBuilder.for_app(app)
-                        .get_case_relationships_for_case_type(case_type))
+        case_property_mapping = builder.get_case_property_map([case_type])
+
+        parent_types = builder.get_case_relationships_for_case_type(case_type)
```

Comment:
This is the main change - instead of using `ParentCasePropertyBuilder.for_app` which fetches all the 'case sharing' apps in the domain, we construct the builder with only the current app. We can also reuse it to get the parent types (not sure if there's any performance benefit there but we don't have to re-construct it (the previous version re-constructed the builder which fetched the 'case sharing' apps AGAIN).